trade-finance-manager-ui/server/controllers/utilisation-reports/helpers/edit-payment-helper.test.ts

```diff
@@ -0,0 +1,807 @@
+        feeRecords,
+      };
+
+      // Act
+      const viewModel = getEditPaymentViewModel(editPaymentResponseBody, reportId, paymentId);
+
+      // Assert
+      expect(viewModel.feeRecords).toHaveLength(feeRecordReportedFees.length);
+      viewModel.feeRecords.forEach(({ reportedFees }, index) => expect(reportedFees).toEqual(formattedReportedFees[index]));
+    });
+
+    it('sets the render view model feeRecords reportedPayments to the edit payment details response feeRecords sorted and formatted reportedPayments', () => {
+      // Arrange
+      const feeRecordReportedPayments: CurrencyAndAmount[] = [
+        { currency: 'GBP', amount: 100 },
+        { currency: 'EUR', amount: 50 },
+      ];
+      const formattedReportedPayments: SortedAndFormattedCurrencyAndAmount[] = [
+        { formattedCurrencyAndAmount: 'GBP 100.00', dataSortValue: 1 },
+        { formattedCurrencyAndAmount: 'EUR 50.00', dataSortValue: 0 },
+      ];
+      const feeRecords = feeRecordReportedPayments.map((reportedPayments) => ({ ...aFeeRecord(), reportedPayments }));
+
+      const editPaymentResponseBody: GetEditPaymentDetailsResponseBody = {
+        ...aEditPaymentDetailsResponseBody(),
+        feeRecords,
+      };
+
+      // Act
+      const viewModel = getEditPaymentViewModel(editPaymentResponseBody, reportId, paymentId);
+
+      // Assert
+      expect(viewModel.feeRecords).toHaveLength(feeRecordReportedPayments.length);
+      viewModel.feeRecords.forEach(({ reportedPayments }, index) => expect(reportedPayments).toEqual(formattedReportedPayments[index]));
+    });
+
+    it("sets the render view model feeRecords checkboxId to 'feeRecordId-' followed by the fee record id", () => {
+      // Arrange
+      const feeRecordIds = [1, 2];
+      const feeRecords = feeRecordIds.map((id) => ({ ...aFeeRecord(), id }));
+
+      const editPaymentResponseBody: GetEditPaymentDetailsResponseBody = {
+        ...aEditPaymentDetailsResponseBody(),
+        feeRecords,
+      };
+
+      // Act
+      const viewModel = getEditPaymentViewModel(editPaymentResponseBody, reportId, paymentId);
+
+      // Assert
+      expect(viewModel.feeRecords).toHaveLength(feeRecordIds.length);
+      viewModel.feeRecords.forEach(({ checkboxId }, index) => expect(checkboxId).toBe(`feeRecordId-${feeRecordIds[index]}`));
+    });
+
+    it('sets the render view model feeRecords isChecked to false for every fee record', () => {
+      // Arrange
+      const feeRecords = [aFeeRecord(), aFeeRecord(), aFeeRecord(), aFeeRecord()];
+
+      const editPaymentResponseBody: GetEditPaymentDetailsResponseBody = {
+        ...aEditPaymentDetailsResponseBody(),
+        feeRecords,
+      };
+
+      // Act
+      const viewModel = getEditPaymentViewModel(editPaymentResponseBody, reportId, paymentId);
+
+      // Assert
+      expect(viewModel.feeRecords).toHaveLength(feeRecords.length);
+      viewModel.feeRecords.forEach(({ isChecked }) => expect(isChecked).toBe(false));
+    });
```

Comment:
The problem is that those two forms are at present completely separate, so you don't even have access to those checkbox ids when you submit the changes to the payment. I'd rather leave that as just a comment on the "Remove selected fee records" ticket as it would be pretty painful to try and wrap everything into the same form

Reply:
I've left a comment on [FN-1719](https://ukef-dtfs.atlassian.net/browse/FN-1719) and will add an AC to it if Oceanne says we should preserve the checkboxes

Reply:
Nice thank you 😄